compute_tools/src/compute.rs

```diff
@@ -754,38 +755,44 @@ LIMIT 100",
             .await?;
             self.ext_remote_paths
                 .set(ext_remote_paths)
-                .expect("ext_remote_paths.set error");
+                .expect("this is the only time we set ext_remote_paths");
+            self.library_index
+                .set(library_index)
+                .expect("this is the only time we set library_index");
         }
         Ok(())
     }
 
-    pub async fn download_extension(&self, ext_name: &str) -> Result<()> {
+    pub async fn download_extension(&self, ext_name: &str, is_library: bool) -> Result<()> {
         match &self.ext_remote_storage {
             None => anyhow::bail!("No remote extension storage"),
             Some(remote_storage) => {
-                // TODO: eliminate useless LOAD Library calls to this function (if possible)
-                // not clear that we can distinguish between useful and useless
-                // library calls better than the below code
-                let ext_name = ext_name.replace(".so", "");
+                let mut real_ext_name = ext_name.to_string();
+                if is_library {
+                    real_ext_name = real_ext_name.replace(".so", "");
+                    real_ext_name =
+                        self.library_index.get().expect("oncelock err")[&real_ext_name].clone();
```

Comment:
Is the idea to panic if the library_index is yet to be downloaded, or should it always be populated at this point? If so, suggestion for the panic message:

```suggestion
                        self.library_index.get().expect("must have already downloaded the library_index")[&real_ext_name].clone();
```

If it should be downloaded on-demand, perhaps `tokio::sync::Once` would help.

Reply:
> Is the idea to panic if the library_index is yet to be downloaded, or should it always be populated at this point? If so, suggestion for the panic message:

it is guaranteed to already be populated at this point

Reply:
thanks, that is indeed a better expect message :)